index.js

```diff
@@ -145,16 +145,34 @@ function $asInteger (i) {
 }
 
 function $asNumber (i) {
+  if (i === 0) return '0'
+  if (!i || i === true) {
+    throw new Error('Cannot coerce to number')
```

Comment:
can you move this `throw` in its own function? Maybe we can avoid the slowdown.

Reply:
Good point

Reply:
The benchmarks don't change. Probably because v8 applies the inline function here.

```
$ node benchmark.js 
$asNumber x 59,491,275 ops/sec ±2.15% (76 runs sampled)
$asNumberOld x 88,423,066 ops/sec ±1.88% (77 runs sampled)
$asBoolean x 310,544,231 ops/sec ±1.98% (77 runs sampled)
$asBooleanOld x 311,224,120 ops/sec ±1.67% (76 runs sampled)
```

Reply:
Remove this `if`.